### PR TITLE
feat(protocol-designer): create StepSummary component

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

This PR creates and implements the new `StepSummary` component to be shown under the deck map in `ProtocolSteps`. The component produces different messages based on all step types and formats them with inline tags and various StyledText styles.

TODO:
- [ ] add test coverage
- [ ] add "note" text once rename step modal is wired up ([reference](https://github.com/Opentrons/opentrons/pull/16437/files#diff-b6e3ca58ba74f1c21f42965649d5be8790919ccbbfa4fc3c43be271daa066656))

Closes AUTH-857

## Test Plan and Hands on Testing

- Create or import a protocol with a variety of step types
- In protocol edit view, click steps to select and verify that new `StepSummary` component displays step info at underneath the deck map

## Changelog

- create new `StepSummary` component and wire up in `ProtocolSteps`
- add translations

## Review requests

- see test plan

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    default:
+      stepSummaryContent = null
+  }
+
+  return stepSummaryContent != null ? (
+    <Flex
+      backgroundColor={`${COLORS.grey30}`}
+      padding={SPACING.spacing12}
+      borderRadius={BORDERS.borderRadius4}
+      width="max-content"
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepSummary.tsx]:
```suggestion
      width={FLEX_MAX_CONTENT}
```

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+        </Flex>
+      )
+      break
+    default:
+      stepSummaryContent = null
+  }
+
+  return stepSummaryContent != null ? (
+    <Flex
+      backgroundColor={`${COLORS.grey30}`}
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepSummary.tsx]:
```suggestion
      backgroundColor={COLORS.grey30}
```

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+            ) : null}
+            <StyledTrans
+              i18nKey="protocol_steps:heater_shaker.active.latch"
+              tagText={t(
+                latchOpen
+                  ? 'protocol_steps:heater_shaker.latch.open'
+                  : 'protocol_steps:heater_shaker.latch.closed'
+              )}
+            />
+          </Flex>
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepSummary.tsx:317]: oh ya, these timer fields (minutes and seconds) are going to be combined at some point, similarly to what you refactored for the pause form.

https://opentrons.atlassian.net/browse/AUTH-864

File: protocol-designer/src/pages/Designer/ProtocolSteps/index.tsx [hunk 4]
```diff
@@ -73,11 +83,18 @@ export function ProtocolSteps(): JSX.Element {
             }}
           />
         ) : null}
-        {deckView === leftString ? (
-          <DeckSetupContainer tab="protocolSteps" />
-        ) : (
-          <OffDeck tab="protocolSteps" />
-        )}
+        <Flex
+          width="751px"
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/index.tsx]: nit 
```suggestion
          width="46.9375rem"
```

File: protocol-designer/src/pages/Designer/ProtocolSteps/__tests__/ProtocolSteps.test.tsx [hunk 5]
```diff
@@ -40,6 +41,7 @@ describe('ProtocolSteps', () => {
 
   it('renders each component in ProtocolSteps', () => {
     render()
+    screen.debug()
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/__tests__/ProtocolSteps.test.tsx]:
```suggestion
```

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx [hunk 6]
```diff
@@ -139,6 +139,7 @@ export function StepFormToolbox(props: StepFormToolboxProps): JSX.Element {
   return (
     <>
       <Toolbox
+        position="relative"
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx]:
```suggestion
      position={POSITION_RELATIVE}
```

**Review by jerader — COMMENTED** [review]
off-deck view which is only visible for a move labware step i think, looks like this:

Screenshot 2024-10-15 at 16 05 48

Otherwise, this is looking great! nice work! love to see the command summaries come to life 🤩

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx (hunk 6 above)
Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx]: oops!

**Comment by ncdiehl11** [comment]
> off-deck view which is only visible for a move labware step i think, looks like this:
> 
> Screenshot 2024-10-15 at 16 05 48
> Otherwise, this is looking great! nice work! love to see the command summaries come to life 🤩

Thanks! I will figure out what I broke here. Likely an empty fragment instead of a div.

**Review by jerader — COMMENTED** [review]
one more comment, i think we want the command summaries to update on hover instead of on selected, similarly to how the deck map also updates on hover.


https://github.com/user-attachments/assets/fc48760b-ddc4-401e-98f1-39535c3640cd

**Comment by ncdiehl11** [comment]
> one more comment, i think we want the command summaries to update on hover instead of on selected, similarly to how the deck map also updates on hover.
> 
>  Screen.Recording.2024-10-16.at.08.41.48.mov

Thanks, missed this! I will update the selector to get the hovered step ID

**Comment by jerader** [comment]
> > one more comment, i think we want the command summaries to update on hover instead of on selected, similarly to how the deck map also updates on hover.
> > Screen.Recording.2024-10-16.at.08.41.48.mov
> 
> Thanks, missed this! I will update the selector to get the hovered step ID

@ncdiehl11 no worries! i guess it should work similarly to the deck map so it should show on hover but also show on selected if you are not hovering over anything.

**Review by jerader — APPROVED** [review]
woo! awesome work 🤩 🎉 lgtm

**Pull request merged**